acceptance/openstack/networking/v2/extensions/agents/agents_test.go

```diff
@@ -92,3 +94,54 @@ func TestAgentsRUD(t *testing.T) {
 	err = agents.Delete(client, allAgents[0].ID).ExtractErr()
 	th.AssertNoErr(t, err)
 }
+
+func TestBGPAgentRUD(t *testing.T) {
+	clients.RequireAdmin(t)
+
+	client, err := clients.NewNetworkV2Client()
+	th.AssertNoErr(t, err)
+
+	// List BGP Agents
+	listOpts := &agents.ListOpts{
+		AgentType: "BGP Dynamic Routing Agent",
+	}
+	allPages, err := agents.List(client, listOpts).AllPages()
+	th.AssertNoErr(t, err)
+
+	allAgents, err := agents.ExtractAgents(allPages)
+	th.AssertNoErr(t, err)
+
+	t.Logf("Retrieved BGP agents")
+	tools.PrintResource(t, allAgents)
+
+	// Create a BGP Speaker
+	bgpSpeaker, err := spk.CreateBGPSpeaker(t, client)
+	th.AssertNoErr(t, err)
+
+	// List the BGP Agent that accommodate the BGP Speaker
+	pages, err := agents.ListDRAgentHostingBGPSpeakers(client, bgpSpeaker.ID).AllPages()
+	th.AssertNoErr(t, err)
+	bgpAgents, err := agents.ExtractAgents(pages)
+	th.AssertNoErr(t, err)
+	th.AssertEquals(t, len(bgpAgents), 1)
```

Comment:
Is it expected that `configurations.bgp_speakers` increases on the agent? Perhaps we could add a test.

Reply:
@mandre 

Sorry for the delay. I have updated the MR. Now it is using Aegnt.Configurations["bgp_speakers"] to assert the bgp speaker scheduling. 

I have to wait 30 seconds between every test. I think this is caused by memcache.